frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareAppStoreVpp/SoftwareVppForm/SoftwareVppForm.tsx

```diff
@@ -0,0 +1,318 @@
+    isValid: !!softwareVppForEdit, // Disables submit before VPP to add is selected
+  });
+
+  const onFormSubmit = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    onSubmit(formData);
+  };
+
+  const onSelectApp = (app: IVppApp) => {
+    if ("selectedApp" in formData) {
+      const newFormData = {
+        ...formData,
+        selectedApp: app,
+        selfService:
+          app.platform === "ios" || app.platform === "ipados"
+            ? false
+            : formData.selfService,
+      };
+      setFormData(newFormData);
+      setFormValidation(generateFormValidation(newFormData));
+    }
+  };
+
+  const onSelectTargetType = (value: string) => {
+    const newData = { ...formData, targetType: value };
+    setFormData(newData);
+    setFormValidation(generateFormValidation(newData));
+  };
+
+  const onSelectCustomTargetOption = (value: string) => {
+    const newData = { ...formData, customTarget: value };
+    setFormData(newData);
+  };
+
+  const onSelectLabel = ({ name, value }: { name: string; value: boolean }) => {
+    const newData = {
+      ...formData,
+      labelTargets: { ...formData.labelTargets, [name]: value },
+    };
+    setFormData(newData);
+    setFormValidation(generateFormValidation(newData));
+  };
+
+  const isSubmitDisabled = !formValidation.isValid;
+
+  const renderSelfServiceContent = (platform: string) => {
+    if (platform !== "ios" && platform !== "ipados") {
+      return (
+        <Checkbox
+          value={formData.selfService}
+          onChange={(newVal: boolean) =>
+            setFormData({ ...formData, selfService: newVal })
+          }
+          className={`${baseClass}__self-service-checkbox`}
+          tooltipContent={
+            <>
+              End users can install from <b>Fleet Desktop</b> {">"}{" "}
+              <b>Self-service</b>.
+            </>
+          }
+        >
+          Self-service
+        </Checkbox>
+      );
+    }
+    return null;
+  };
+
+  const renderContent = () => {
+    if (softwareVppForEdit) {
+      return (
+        <div className={`${baseClass}__form-fields`}>
+          <FileUploader
+            canEdit={false}
+            graphicName={"app-store"}
+            message=".pkg, .msi, .exe, .deb, or .rpm"
+            className={`${baseClass}__file-uploader`}
+            fileDetails={{ name: softwareVppForEdit.name, platform: "macOS" }}
+          />
+          <TargetLabelSelector
+            selectedTargetType={formData.targetType}
+            selectedCustomTarget={formData.customTarget}
+            selectedLabels={formData.labelTargets}
+            customTargetOptions={CUSTOM_TARGET_OPTIONS}
+            className={`${baseClass}__target`}
+            onSelectTargetType={onSelectTargetType}
+            onSelectCustomTarget={onSelectCustomTargetOption}
+            onSelectLabel={onSelectLabel}
+            labels={labels || []}
+          />
+          {renderSelfServiceContent("macOS")}
```

Comment:
Same note here as with the uploader.

Reply:
! omg, nice catch, this is a blocker, we need to know the type to render this conditionally

Reply:
Realized the API is actually returning platform, but it was missing from the FE interface `IAppStoreApp`. Added, unblocked